### PR TITLE
HEC-1047 chargeable for corporation tax page

### PR DESCRIPTION
**Pull request by meghana-viswanath**
Details in ticket -> https://jira.tools.tax.service.gov.uk/browse/HEC-1047

### CODE REVIEW

File: app/uk/gov/hmrc/hecapplicantfrontend/controllers/CompanyDetailsController.scala [hunk 1]
```diff
@@ -173,9 +176,69 @@ class CompanyDetailsController @Inject() (
...
+          val chargeableForCT = request.sessionData.userAnswers.fold(_.chargeableForCT, _.chargeableForCT)
+          val form = {
+            val emptyForm = CompanyDetailsController.chargeableForCTForm(YesNoAnswer.values)
+            chargeableForCT.fold(emptyForm)(emptyForm.fill)
+          }
+          Ok(
+            chargeableForCTPage(
+              form = form,
+              back = journeyService.previous(routes.CompanyDetailsController.chargeableForCorporationTax()),
+              date = formattedDate(latestAccountingPeriod.endDate),
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/controllers/CompanyDetailsController.scala:194]: this should use `TimeUtils.govDisplayFormat` to make sure this gets translated when we come to do the welsh translation of the service - there's an option to pass in `LocalDate` to the template to have a more meaningful input parameter to it as well.

File: app/uk/gov/hmrc/hecapplicantfrontend/services/JourneyService.scala [hunk 2]
```diff
@@ -324,8 +346,16 @@ object JourneyServiceImpl {
       case _ => true
     }
 
+  private def checkCompanyData(
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/services/JourneyService.scala:349]: maybe something like `checkCompanyDataComplete` would make it clearer without having to look at the implementation?

File: app/uk/gov/hmrc/hecapplicantfrontend/views/ChargeableForCT.scala.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        attributes = Map("aria-describedby" -> "company-name")
+      )),
+      idPrefix = Some(key),
+      name = key,
+      items = radioOptions,
+      errorMessage = form.error(key).map(e => ErrorMessage(
+        content = Text(messages(s"$key.${e.message}"))
+      ))
+    ))
+    <p class="govuk-body"><a class="govuk-link" href="@{appConfig.accountingPeriodsGuidanceUrl}">@{messages(s"$key.link")}</a></p>
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/views/ChargeableForCT.scala.html:61]: @nubz should we be using the `HmrcNewTabLink` component here for the link?

Comment (nubz) [reply app/uk/gov/hmrc/hecapplicantfrontend/views/ChargeableForCT.scala.html:61]: yes this should be an external link into a new tab

File: test/uk/gov/hmrc/hecapplicantfrontend/services/JourneyServiceImplSpec.scala [hunk 4]
```diff
@@ -1072,6 +1072,135 @@ class JourneyServiceSpec extends AnyWordSpec with Matchers with MockFactory with
 
         }
 
+        "chargeable for CT page" should {
+          val chargeableForCorporationTaxRoute = routes.CompanyDetailsController.chargeableForCorporationTax()
+          val date                             = LocalDate.now
+
+          "throw" when {
```
Comment (priyankadhiman3006) [inline test/uk/gov/hmrc/hecapplicantfrontend/services/JourneyServiceImplSpec.scala:1079]: throw error?

**Review by priyankadhiman3006 — APPROVED** [review]

**Pull request merged**